### PR TITLE
Remove all trigger words except for "weather"

### PR DESCRIPTION
**Pull request by tommytommytommy**
The weather IA is overtriggering and producing some odd results when we pass Forecast an extracted location that it does not fail on. To try to address this, I would like to limit the keyword triggering for this IA to only "weather," and then handle all other potential weather/forecast-related queries with an internal trigger that looks at search results to infer the user's intent.

### CODE REVIEW

**Comment by MrChrisW** [comment]
Hey @tommytommytommy ! Better update the [tests](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/t/Forecast.t#L82) as well :+1:

**Comment by tommytommytommy** [comment]
@MrChrisW Thanks for the heads-up! I commented out the tests and triggers in the IA instead of deleting them because they might be good examples for others that are exploring what they can do with triggers in their own Spice. If it's preferable, I am happy to delete all of the commented out sections to clean up the code.

File: lib/DDG/Spice/Forecast.pm [hunk 1]
```diff
@@ -12,28 +12,30 @@ secondary_example_queries "weather 12180";
...
+
+my @forecast_words = qw(weather);
+my @triggers = (@forecast_words);
 triggers startend => @triggers;
 
 spice from => '([^/]*)/?([^/]*)';
 spice to => 'http://forecast.io/ddg?apikey={{ENV{DDG_SPICE_FORECAST_APIKEY}}}&q=$1&callback={{callback}}';
 
+# skip synonyms for "weather" because of overtriggering
+my @weather_synonyms =qw(forecast);
```
Comment (moollaza) [inline lib/DDG/Spice/Forecast.pm]: hehe maybe we should change the name of the IA too, then?

File: lib/DDG/Spice/Forecast.pm [hunk 2]
```diff
@@ -12,28 +12,30 @@ secondary_example_queries "weather 12180";
...
-        push @temps_triggers, $temp . ' ' . $conn;    # temperature for, temp at
-        foreach my $desc (@descriptors) {
-            push @temps_triggers, join(' ', $desc, $temp, $conn), $desc . ' ' . $temp;    # current temperature in; current temp
-        }
-    }
-}
-
-my @forecast_words = qw(forecast forcast weather);
-my @triggers = (@forecast_words, @temps_triggers, 'meteo');
+# my @temperatures = qw(temp temperature);
```
Comment (moollaza) [inline lib/DDG/Spice/Forecast.pm]: Might as well remove this, we can always revert the commit to get it back.

File: t/Forecast.t [hunk 3]
```diff
@@ -12,24 +12,24 @@ my $loc = test_location('de');
...
-    DDG::Request->new(
-        query_raw => 'forecast',
-        location => $loc
-    ) => test_spice(
-        "/js/spice/forecast/" . uri_escape_utf8(${\$loc->loc_str}) . "/current",
-        call_type => 'include',
-        caller => 'DDG::Spice::Forecast',
-        is_cached => 0
-    ),
+    #     DDG::Request->new(
```
Comment (moollaza) [inline t/Forecast.t]: Same here, let's just remove this to simplify things. The docs and (hopefully) all other IA's should be providing good examples.

File: lib/DDG/Spice/Forecast.pm (hunk 1 above)
Comment (tommytommytommy) [inline lib/DDG/Spice/Forecast.pm]: Haha...well, since we're still using proper noun Forecast as the provider, let's leave it for now :)

**Comment by tommytommytommy** [comment]
@moollaza Comments removed!

File: lib/DDG/Spice/Forecast.pm (hunk 1 above)
Comment (moollaza) [inline lib/DDG/Spice/Forecast.pm]: :+1:

**Comment by moollaza** [comment]
@tommytommytommy LGTM :+1:

**Pull request merged**